src/Common/Auth/OpenIDConnect/Repositories/AccessTokenRepository.php

```diff
@@ -71,4 +100,30 @@ public function getNewToken(ClientEntityInterface $clientEntity, array $scopes,
 
         return $accessToken;
     }
+
+    public function getActiveTokensForUser($clientId, $userUuid)
+    {
+        // note user_id is the STRING representation of the uuid, not the binary representation
+        $sql = "SELECT * FROM api_token WHERE user_id = ? AND client_id = ? AND expiry > NOW() AND revoked = 0 ";
+        return QueryUtils::fetchRecords($sql, [$userUuid, $clientId]);
+    }
+
+    /**
+     * Retrieves a token record for given database token id.
+     * @param $id The database identifier for the token (see table api_token.id
+     * @return array|null
+     */
+    public function getTokenById($id)
+    {
+        $sql = "SELECT * FROM api_token WHERE id = ? ";
+        $records = QueryUtils::fetchRecords($sql, [$id]);
+        return $records[0] ?? null;
+    }
+
+    public function getTokenByToken($token)
+    {
+        $sql = "SELECT * FROM api_token WHERE id = ? ";
```

Comment:
is this query correct or is it supposed to be WHERE `token` = ...?

Reply:
good catch, wonder why it was working haha.